### DISCUSSION TITLE
Corrupted records in export to CSV

### DISCUSSION THREAD

**Discussion by monotasker** [MESH-Research/knowledge-commons-repository#79] [General]
@mikethicke: I've identified one record that seems to be involved in several corrupted rows of data: https://hcommons.org/deposits/item/hc:44597/

The record appears fine on its landing page and produces normal mods xml. In the csv the record is normal up to the abstract field. The abstract is cut off after the sentence "It was the breaking away from this idea that was perhaps the most important aspect of the works of both Immanuel Kant (1724-1804) and David Hume (1711-1776)." The rest of the abstract text is then spread over the first few fields of the next several rows. Within each of those rows it looks like the text is divided at commas. But I can't immediately see what in the text is triggering the skip to a new row each time. 

The first thing I noticed was the unescaped single quotes, but they don't seem to coincide with the breaks.

**Comment by monotasker** [comment]
Okay, I looked in the Fedora output. The abstract has newlines ('\n') that the csv is interpreting as the beginning of a new record. Is that an easy fix in your csv export script?

**Comment by monotasker** [comment]
hc:45615 is also having the abstract split on commas and spread across subsequent fields. Not sure with this one why the split happens where it does, since there is one comma that *isn't* interpreted as a field break early in the abstract string.

**Reply by monotasker** [reply]
After a little more digging, here too the split begins at the first comma after the sequence

`priests: \\&quot;When manâ\x80\x99s life`

**Comment by monotasker** [comment]
hc:44849 has a similar split of the abstract. The splitting on commas begins after the sequence of characters:

`of \\&quot; symbolic violence .\\&quot;`

**Comment by monotasker** [comment]
hc:43939 also splits the abstract and starts splitting on the first comma after the sequence 

`meaning of \\&quot;social\\&quot; and \\&quot;collective power\\&quot; is`

So is there something about the way those slash escapes and html entities combine that throws things off?

**Reply by mikethicke** [reply]
Yeah there might be some issues with double escaping. The php function that writes to CSVs is somewhat smart about escaping but it looks like some extra work needs to be done.

**Reply by monotasker** [reply]
On double checking, I think the double escaping *may* be an artifact of the way python strings are represented in command line output. The actual character may be a single slash.

**Comment by monotasker** [comment]
We ended up resolving all of this by exporting to JSON instead of csv. (Exporting json strings as csv fields created headaches when we tried to parse with python.)

**Comment by bjr70** [comment]
If this is resolved should we close this discussion?
